File: src/main/java/br/com/leonardoferreira/jirareport/domain/response/BoardDetailsResponse.java
package br.com.leonardoferreira.jirareport.domain.response;

import br.com.leonardoferreira.jirareport.domain.ImpedimentType;
import br.com.leonardoferreira.jirareport.domain.vo.DynamicFieldConfig;
import java.util.List;
import lombok.Data;
import lombok.ToString;

@Data
@ToString(of = "name")
public class BoardDetailsResponse {

    private Long externalId;

    private String name;

    private String startColumn;

    private String endColumn;

    private List<String> fluxColumn;

    private List<String> ignoreIssueType;

    private String epicCF;

    private String estimateCF;

    private String systemCF;

    private String projectCF;

    private Boolean calcDueDate;

    private Boolean ignoreWeekend;

    private ImpedimentType impedimentType;

    private List<String> impedimentColumns;

    private List<DynamicFieldConfig> dynamicFields;

}

File: src/main/kotlin/br/com/jiratorio/domain/response/board/BoardDetailsResponse.kt
package br.com.jiratorio.domain.response.board

import br.com.jiratorio.domain.entity.DynamicFieldConfig
import br.com.jiratorio.domain.impediment.ImpedimentType

data class BoardDetailsResponse(
    val id: Long,
    val externalId: Long? = null,
    val name: String? = null,
    val startColumn: String? = null,
    val endColumn: String? = null,
    val fluxColumn: List<String>? = null,
    val ignoreIssueType: List<String>? = null,
    val epicCF: String? = null,
    val estimateCF: String? = null,
    val systemCF: String? = null,
    val projectCF: String? = null,
    val ignoreWeekend: Boolean? = null,
    val dueDateCF: String? = null,
    val dueDateType: String? = null,
    val impedimentType: ImpedimentType? = null,
    val impedimentColumns: List<String>? = null,
    val touchingColumns: List<String>? = null,
    val waitingColumns: List<String>? = null
)
